src/wasm-v8.cc

```diff
@@ -1453,23 +1463,32 @@ auto Module::deserialize(Store* store_abs, const vec<byte_t>& serialized) -> own
 
 
 // TODO(v8): do better when V8 can do better.
-template<> struct implement<Shared<Module>> { using type = vec<byte_t>; };
+auto impl(Shared<Module>* x) -> vec<byte_t>* {
```

Comment:
IIUC, the `implement` template no longer works for this because it requires a reinterpret_cast? Could that be fixed by introducing a SharedImpl subclass as well?

Reply:
You'd need to make `vec<byte_t>` derive from the subclass too.

Reply:
No, that's not quite right. We could also have `SharedImpl` derive from both `Shared<Module>` and from `vec<byte_t>`.

Reply:
Done. I made `SharedImpl<T>` derive from `vec<byte_t>` but when you add `SharedImpl<T != Module>` you'll probably want to make that optional.